src/main/java/appeng/tile/storage/TileChest.java

```diff
@@ -1043,4 +1002,59 @@ public FluidStack drain( final int maxDrain, final boolean doDrain )
 		}
 	}
 
+	private class InputInventoryFilter implements IAEItemFilter
+	{
+		@Override
+		public boolean allowExtract( IItemHandlerModifiable inv, int slot, int amount )
+		{
+			return false;
+		}
+
+		@Override
+		public boolean allowInsert( IItemHandlerModifiable inv, int slot, ItemStack stack )
+		{
+			if( isPowered() )
+			{
+				try
+				{
+					final IMEInventory<IAEItemStack> cell = getHandler( StorageChannel.ITEMS );
+					if( cell != null )
+					{
+						final IAEItemStack returns = cell.injectItems( AEApi.instance().storage().createItemStack( stack ), Actionable.SIMULATE, mySrc );
+						return returns == null || returns.getStackSize() != stack.getCount();
+					}
+				}
+				catch( final ChestNoHandler ignored )
+				{
+				}
+			}
+			return false;
+		}
+	}
+
+	private class CellInventoryFilter implements IAEItemFilter
+	{
+
+		@Override
+		public boolean allowExtract( IItemHandlerModifiable inv, int slot, int amount )
+		{
+			return true;
+		}
+
+		@Override
+		public boolean allowInsert( IItemHandlerModifiable inv, int slot, ItemStack stack )
+		{
+			if( AEApi.instance().registries().cell().getCellInventory( stack, TileChest.this, StorageChannel.ITEMS ) != null )
+			{
+				return true;
+			}
+			if( AEApi.instance().registries().cell().getCellInventory( stack, TileChest.this, StorageChannel.FLUIDS ) != null )
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1126)